### DISCUSSION TITLE
prefer `aria-`,  `data-` over css classes in element picker

### DISCUSSION THREAD

**Discussion by scarf005** [uBlockOrigin/uBlock-issues#3149] [Ideas] [answered]
[original URL (twitter)](https://twitter.com/jojoldu/status/1763140848555077786)

| as-is | to-be |
| - | - |
| ![image](https://github.com/uBlockOrigin/uBlock-issues/assets/54838975/b36dc25c-3f5b-4457-a2b1-311c3e38c7f5) | ![image](https://github.com/uBlockOrigin/uBlock-issues/assets/54838975/abc8e4fa-fd96-4438-9d01-38e941f8ab63) |

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/54838975/9e09feb9-de1f-4307-823e-61d787eb50da)

due to css minifier and obfuscators, element picker often picks hashed css classes.
this has following disadvantages:
- it's vulnerable to hash changes.
- it's hard to understand cryptic filters

making the filter to prefer human-readable properties like [`aria-label`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes) and [`data-testid`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*) over css classes (like how id has higher precedence over classes) would make it easier to zap elements.

**Comment by gorhill** [comment, marked as answer]
The picker can't tell whether a class name is meaningful or just random hashes as a result of server-side compiler.

The solution is to provide more options for users to decide what should be used -- something I have entertained since a long while now, but this requires work. I had been thinking that for each entry there would be a collapsible/expandable list of checkboxes about which tag/class/names/attributes should be used for that entry.

**Reply by scarf005** [reply]
yeah, providing radio group would work well.

**Comment by garry-ut99** [comment]
Duplicate of this comment :
- https://github.com/uBlockOrigin/uBlock-issues/issues/2046#issuecomment-1260897868

Also similiar to / related to :
- https://github.com/uBlockOrigin/uBlock-issues/issues/2046
- https://github.com/uBlockOrigin/uBlock-issues/issues/1052
- https://github.com/gorhill/uBlock/issues/1736
